### PR TITLE
DATAUP-298 Tests!

### PR DESCRIPTION
**Pull request by eamahanna**
# Description of PR purpose/changes

This is the final ticket (finger crossed) for DATAUP-298. It adds tests for the `filePathWidget` and `fieldTableCellWidget`.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-289
- [ ] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
Execute `make test-frontend-unit`

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the travis build passes)

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: test/unit/spec/common/cellComponents/fieldTableCellWidget-Spec.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
+/*global beforeEach */
+/*global define*/ // eslint-disable-line no-redeclare
+/*global describe, expect, it*/
+/*global beforeAll, afterAll*/
+/*jslint white: true*/
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/fieldTableCellWidget-Spec.js:5]: These are all now covered by the `.eslintrc.yml` file in `test/`

File: test/unit/spec/common/cellComponents/filePathWidget-Spec.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                workspaceId: workspaceId,
+                initialParams: model.getItem('params'),
+            });
+        });
+
+        afterEach(() => {
+            $('body').empty();
+        });
+
+        it('has a factory which can be invoked', function() {
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/filePathWidget-Spec.js:77]: Should this description match the test more accurately?

File: test/unit/spec/common/cellComponents/filePathWidget-Spec.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            expect(filePathWidget.bus).toBeDefined();
+        });
+
+        it('should start and render itself', () => {
+            return filePathWidget.start({
+                node: node,
+                appSpec: spec,
+                parameters: parameters,
+            }).then(() => {
+                expect(node.innerHTML).toContain('File Paths');
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/filePathWidget-Spec.js:93]: I'd just put in an array of things that you want to look for, and then cycle through them with a `for` loop, i.e.

```js
const contents = ['File Paths', 'kb-file-path__table', '...'];
contents.forEach( (item) => {
  expect(node.innerHTML).toContain(item);
});
```

File: test/unit/spec/common/cellComponents/filePathWidget-Spec.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                expect(node.innerHTML).toContain('kb-file-path__table');
+                expect(node.innerHTML).toContain('kb-file-path__table_row');
+                expect(node.innerHTML).toContain('kb-file-path__file_number');
+                expect(node.innerHTML).toContain('1');
+                expect(node.innerHTML).toContain('fastq_rev_staging_file_name');
+                expect(node.innerHTML).toContain('sra_staging_file_name');
+                expect(node.innerHTML).toContain('fa fa-trash-o fa-lg');
+                expect(node.innerHTML).toContain('Add Row');
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/filePathWidget-Spec.js:101]: You have imported jQuery, so why not do some more exact queries? e.g. 

```js
expect($('.kb-file-path__file_number').innerHTML).toContain('1');
expect($('.sra_staging_file_name').innerHTML).toContain('whatever');
```

You can even still use a nice `forEach` loop, e.g.
```js
const contents = [
  { selector: '.kb-file-path__file_number', content: '1', },
  { selector: '.sra_staging_file_name', content: 'whatever', },
  ...
];

contents.forEach( (item) => {
  expect( $( item.selector ).innerHTML ).toContain( item.content );
});
```

File: test/unit/spec/common/cellComponents/filePathWidget-Spec.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            });
+        });
+
+        it('should add a row when Add Row button is clicked', () => {
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/filePathWidget-Spec.js:105]: 👍

File: test/unit/spec/common/cellComponents/filePathWidget-Spec.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                let preClickNumberOfRows = $('tr').length;
+                expect(preClickNumberOfRows).toEqual(1);
+                console.log($(node).find('.kb-file-path__button--add_row btn btn__text'));
+                $(node).find('.kb-file-path__button--add_row btn btn__text').click();
+                let postClickNumberOfRows = $('tr').length;
+                expect(postClickNumberOfRows).toEqual(2);
+            });
+        });
+
+        it('should delete a row when trashcan button is clicked', () => {
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/filePathWidget-Spec.js:120]: Worth writing a test with two rows and ensuring that the correct row is deleted?

File: test/unit/spec/common/cellComponents/fieldTableCellWidget-Spec.js [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+            expect(mockFieldWidget.stop).toBeDefined();
+            expect(mockFieldWidget.bus).toBeDefined();
+        });
+
+        it('has a method start which returns the correct object', () => {
+            return mockFieldWidget.start({
+                node: node
+            }).then(() => {
+                expect(node.innerHTML).toContain('kb-field-cell__cell_label');
+                expect(node.innerHTML).toContain('kb-field-cell__input_control');
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/fieldTableCellWidget-Spec.js:145]: Can you use jQuery and do some more precise tests here?

File: test/unit/spec/common/cellComponents/filePathWidget-Spec.js (hunk 3 above)
Comment (eamahanna) [reply test/unit/spec/common/cellComponents/filePathWidget-Spec.js:87]: good call

File: test/unit/spec/common/cellComponents/filePathWidget-Spec.js (hunk 4 above)
Comment (eamahanna) [reply test/unit/spec/common/cellComponents/filePathWidget-Spec.js:95]: I am just checking the constants such as `Add Row` and the classes exist. I don't think testing the text matters too much.

File: kbase-extension/scss/partials/_bulkImportCell.scss [hunk 8]
```diff
@@ -236,6 +236,9 @@ $button-outdated: rgb(247, 155, 34);
 
 .kb-file-path {
   /* .kb-file-path-table-row */
+  &__table {
+    width: 100%
```
Comment (eamahanna) [inline kbase-extension/scss/partials/_bulkImportCell.scss:240]: Fixed a style issue found while working with Mallory.

File: kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js [hunk 9]
```diff
@@ -218,8 +218,8 @@ define([
             });
         }
 
-        function addRow(){
-            $(`.${cssBaseClass}__table`).append(
+        function addRow(e){
+            $(e.target).prev('table').append(
```
Comment (eamahanna) [inline kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js:222]: @briehl This was causing the row issue. The selector returned all tables. So when there was more than one table in the narrative, a row was added to each.

File: nbextensions/bulkImportCell/testAppObj.json [hunk 10]
```diff
@@ -228,28 +228,6 @@
             "ui_class": "parameter",
             "ui_name": "Reverse/Right FASTA/FASTQ File Path"
           },
-          {
```
Comment (eamahanna) [inline nbextensions/bulkImportCell/testAppObj.json:231]: Took out this param so Mallory could see the styling with 3 file path params (more similar to what will be shown at the end)

**Review by leia-sefkin — APPROVED** [review]
LGTM! ship it

**Pull request merged**